instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/TemporaryMetricsView.java

```diff
@@ -72,10 +105,22 @@ private static void applyView(
         (BiConsumer<AttributeKey, Object>)
             (key, value) -> {
               if (view.contains(key)) {
-                filtered.put(key, value);
+                // For now, we filter query parameters out of URLs in metrics.
+                if (SemanticAttributes.HTTP_URL.equals(key)) {
+                  filtered.put(
+                      SemanticAttributes.HTTP_URL, removeQueryParamFromUrl(value.toString()));
+                } else {
+                  filtered.put(key, value);
+                }
               }
             });
   }
 
+  private static String removeQueryParamFromUrl(String url) {
+    // Note: Maybe not the most robust, but purely to limit cardinality.
+    int idx = url.lastIndexOf('?');
```

Comment:
perhaps also test for `;` in case url looks like http://www.sample.com/home;jsessionid=BB69F0DEF79877629096AFEFFB53D478?foo=bar

Reply:
Did a quick gander through [allowable syntax](https://www.w3.org/Addressing/URL/url-spec.txt).


- `?` in query parameter MUST be escaped, so I think we're ok there.
- `;` doesn't appear like a valid character.  I added it though.


Wound writing my own search since I'm not sure if there's an index-of-two-characters method.

Reply:
oh no competing specs? 😂

https://datatracker.ietf.org/doc/html/rfc3986#section-3.4

> The characters slash ("/") and question mark ("?") may represent data within the query component

Reply:
also `;` is definitely used in practice, so maybe that's not the right spec to be looking at?

Reply:
1994 vs 2005, you win for more recent.